src/Core/InstanceConfigTrait.php

```diff
@@ -110,11 +120,11 @@ public function setConfig($key, $value = null, $merge = true)
      * $this->getConfig('some-key', 'default-value');
      * ```
      *
-     * @param string|null $key The key to get or null for the whole config.
-     * @param mixed $default The return value when the key does not exist.
-     * @return mixed Configuration data at the named key or null if the key does not exist.
+     * @param string $key The key to get.
+     * @param mixed|null $default The return value when the key does not exist.
```

Comment:
Why the `|null`?  `mixed` covers all types.

Reply:
To explicitly state the nullish part over non null mixed.

Reply:
That seems quite unnecessary and just adds noise IMO.